src/services/default-rule.js

```diff
@@ -42,61 +52,58 @@ module.exports = class DefaultRuleHelper {
 				status: 'ACTIVE',
 				organization_id: defaultOrgId,
 				value: bodyData.requester_field,
-				model_names: {
-					[Op.contains]: [
-						await mentorExtensionQueries.getModelName(),
-						await menteeExtensionQueries.getModelName(),
-					],
-				},
+				model_names: { [Op.contains]: [mentorModelName, menteeModelName] },
 				required: true,
 			}),
 		])
 
+		const [validTargetField, validRequesterField] = await validFieldsPromise
+
 		const errors = []
 
-		if (validTargeField.length === 0) {
-			errors.push({
-				param: 'target_field',
-				msg: `Invalid target_field`,
-			})
+		if (validTargetField.length === 0) {
+			errors.push({ param: 'target_field', msg: 'Invalid target_field' })
 		}
 
 		if (validRequesterField.length === 0) {
-			errors.push({
-				param: 'requester_field',
-				msg: `Invalid requester_field`,
-			})
-		}
-		if (validTargeField.length === 0 || validRequesterField.length === 0) {
-			return errors
-		}
-		if (validTargeField[0]?.data_type != validRequesterField[0]?.data_type) {
-			errors.push({
-				param: 'target_field,requester_field',
-				msg: `Data types of target_field and requester_field should match`,
-			})
-		} else if (
-			common.DEFAULT_RULES.ARRAY_TYPES.includes(validTargeField[0]?.data_type) &&
-			!common.DEFAULT_RULES.VALID_ARRAY_OPERATORS.includes(bodyData.operator)
-		) {
-			errors.push({
-				param: 'operator',
-				msg: `Invalid operator for ARRAY field type`,
-			})
+			errors.push({ param: 'requester_field', msg: 'Invalid requester_field' })
 		}
 
-		if (errors.length !== 0) {
-			return {
-				isValid: false,
-				errors,
+		if (validTargetField.length > 0 && validRequesterField.length > 0) {
+			if (validTargetField[0]?.data_type !== validRequesterField[0]?.data_type) {
+				errors.push({
+					param: 'target_field,requester_field',
+					msg: 'Data types of target_field and requester_field should match',
+				})
+			} else {
+				const operatorValidation = {
```

Comment:
@nevil-mathew checking for first field is enough?

Reply:
We're checking if the target_field and requester_field have the same data_type before this, when we reach this condition both will have same data_type